sources/packages/backend/apps/queue-consumers/src/services/cas-supplier/cas-evaluation-result-processor/cas-active-supplier-not-found-processor.ts

```diff
@@ -0,0 +1,118 @@
+import { Injectable } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { SystemUsersService } from "@sims/services";
+import { CASSupplier, SupplierStatus } from "@sims/sims-db";
+import { ProcessSummary } from "@sims/utilities/logger";
+import {
+  CASEvaluationResult,
+  CASEvaluationStatus,
+} from "../cas-supplier.models";
+import { Repository } from "typeorm";
+import {
+  CASAuthDetails,
+  CASService,
+  CreateSupplierAndSiteResponse,
+} from "@sims/integrations/cas";
+import { CASEvaluationResultProcessor, ProcessorResult } from ".";
+
+/**
+ * Process a student that was not found on CAS.
+ */
+@Injectable()
+export class CASActiveSupplierNotFoundProcessor extends CASEvaluationResultProcessor {
+  constructor(
+    private readonly casService: CASService,
+    private readonly systemUsersService: SystemUsersService,
+    @InjectRepository(CASSupplier)
+    private readonly casSupplierRepo: Repository<CASSupplier>,
+  ) {
+    super();
+  }
+
+  /**
+   * Create the new supplier and site on CAS using the student information.
+   * @param casSupplier student supplier information from SIMS.
+   * @param evaluationResult evaluation result to be processed.
+   * @param auth authentication token needed for possible
+   * CAS API interactions.
+   * @param summary current process log.
+   * @returns processor result.
+   */
+  async process(
+    casSupplier: CASSupplier,
+    evaluationResult: CASEvaluationResult,
+    auth: CASAuthDetails,
+    summary: ProcessSummary,
+  ): Promise<ProcessorResult> {
+    if (evaluationResult.status !== CASEvaluationStatus.NotFound) {
+      throw new Error("Incorrect CAS evaluation result processor selected.");
+    }
+    summary.info(
+      `No active CAS supplier found. Reason: ${evaluationResult.reason}.`,
+    );
+    let result: CreateSupplierAndSiteResponse;
+    try {
+      const address = casSupplier.student.contactInfo.address;
+      result = await this.casService.createSupplierAndSite(auth.access_token, {
+        firstName: casSupplier.student.user.firstName,
+        lastName: casSupplier.student.user.lastName,
+        sin: casSupplier.student.sinValidation.sin,
+        emailAddress: casSupplier.student.user.email,
+        supplierSite: {
+          addressLine1: address.addressLine1,
+          city: address.city,
+          provinceCode: address.provinceState,
+          postalCode: address.postalCode,
+        },
+      });
+      summary.info("Created supplier and site on CAS.");
+    } catch (error: unknown) {
+      summary.error("Error while creating supplier and site on CAS.", error);
+      return { isSupplierUpdated: false };
+    }
+    try {
+      const [submittedAddress] = result.submittedData.SupplierAddress;
+      const now = new Date();
+      const systemUser = this.systemUsersService.systemUser;
+      const updateResult = await this.casSupplierRepo.update(
+        {
+          id: casSupplier.id,
+        },
+        {
```

Comment:
The update happening for found and not found is identical. Can we have a common method to be re-used for the update?

Reply:
Do you mean the below part is identical? I would prefer to leave as it is.

![image](https://github.com/user-attachments/assets/8e7e1045-7a99-4986-8e22-0434873b75a7)

Reply:
My point was conversion of CAS Supplier and CAS Site to SIMS Supplier and SIMS supplier address(and then update). May be something in one of the upcoming CAS tickets.